pkg/managed/istio.go

```diff
@@ -677,17 +677,36 @@ func newServiceEntries(mbPair *types.ModelBindingPair, mc *types.ManagedCluster,
 // issues with WebLogic managed server communication with the admin server.
 func newDestinationRules(mbPair *types.ModelBindingPair, mc *types.ManagedCluster, pods []*v1.Pod) ([]*v1alpha3.DestinationRule, error) {
 	var rules []*v1alpha3.DestinationRule
-	namespaceMap := make(map[string]struct{})
+	namespaceMap := make(map[string]*v1beta1v8o.KubernetesNamespace)
+
+	// Get all the coherence extend ports for each coherence component.
+	// create a map of coherence ports, key is coherence component name
+	var cohPortMap = make(map[string][]uint32)
+	for _, cohCluster := range mbPair.Model.Spec.CoherenceClusters {
+		var ports []uint32
+		for _, p := range cohCluster.Ports {
+			ports = append(ports, uint32(p.Port))
+		}
+		cohPortMap[cohCluster.Name] = ports
+	}
 
 	// get all the namespaces from the binding
 	for _, placement := range mbPair.Binding.Spec.Placement {
-		for _, namespace := range placement.Namespaces {
-			namespaceMap[namespace.Name] = struct{}{}
+		for i, namespace := range placement.Namespaces {
+			namespaceMap[namespace.Name] = &placement.Namespaces[i]
 		}
 	}
-	// create an destination rule for each binding namespace in the given cluster
+
+	// create a destination rule for each binding namespace in the given cluster
 	for _, namespace := range mc.Namespaces {
-		if _, ok := namespaceMap[namespace]; ok {
+		if ns, ok := namespaceMap[namespace]; ok {
+			// If this namespace has a coherence component then get the ports from the map
+			var cohPorts []uint32
+			for _, comp := range ns.Components {
+				if ports, ok := cohPortMap[comp.Name]; ok {
+					cohPorts = ports
```

Comment:
Does this work if a namespace has two coherence components.

Reply:
Namespace cannot have 2 coherence components today (before my change).  It won't work and is not supported

Reply:
I was not aware of this restriction.  We do not disallow it, do we?

Reply:
No, realistically nobody would ever do this with Coherence.  But we should document it.